python/common/org/python/types/List.java

```diff
@@ -337,36 +337,50 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                     if (step < 0) {
                         long start;
                         if (slice.start != null) {
-                            start = Math.min(slice.start.value, this.value.size());
+                            if (slice.start.value < 0) {
+                                start = Math.max((this.value.size() + slice.start.value), -1);
+                            } else {
+                                start = Math.min(slice.start.value, this.value.size());
```

Comment:
Hey @Nasreen123 , it looks great now!! :tada: 
However, while I was testing it, I found another bug that is already present in the previous code, I think this line should be:

    start = Math.min(slice.start.value, this.value.size() - 1);

The only difference is the `-1` at the end.
This will handle correctly the cases of: `x[len(x)::-2]` and `x[len(x)+2::-2]`, which are currently crashing with `IndexOutOfBounds`.

Can you sneak the tests and bugfix for that in this same PR? =)
Thank you!!